src/System.IO.Ports/src/System/IO/Ports/SerialPort.OSX.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.IO;
+
+namespace System.IO.Ports
+{
+    public partial class SerialPort : Component
+    {
+        public static string[] GetPortNames()
+        {
+            List<string> ports = new List<string>();
+
+            foreach (string name in Directory.GetFiles("/dev", "tty.*"))
```

Comment:
nit: since you are using Directory.GetFiles you have an upper-bound for ports, perform the two GetFiles and create ports using this upper-bound as capacity.